wallet_data/src/commonMain/kotlin/net/thechance/mena/wallet/data/di/WalletDataModule.kt

```diff
@@ -1,8 +1,27 @@
 package net.thechance.mena.wallet.data.di
 
+import net.thechance.mena.wallet.data.database.StatementDao
+import net.thechance.mena.wallet.data.database.WalletDatabase
+import net.thechance.mena.wallet.data.database.WalletDatabaseBuilder
+import net.thechance.mena.wallet.data.database.getWalletDataBase
 import org.koin.core.annotation.ComponentScan
 import org.koin.core.annotation.Module
+import org.koin.core.annotation.Single
 
 @Module
 @ComponentScan("net.thechance.mena.wallet.data")
-class WalletDataModule
\ No newline at end of file
+class WalletDataModule{
+    @Single
+    fun provideWalletDatabaseBuilder(): WalletDatabaseBuilder {
+        return WalletDatabaseBuilder()
+    }
+    @Single
+    fun provideWalletDatabase(builder: WalletDatabaseBuilder): WalletDatabase {
+        return getWalletDataBase(builder.getBuilder())
+    }
```

Comment:
What about combining `provideWalletDatabase` and `getWalletDataBase` into one function? i think it's simpler. 

```suggestion
    @Single
    fun provideWalletDatabase(builder: WalletDatabaseBuilder): WalletDatabase {
        return builder.getBuilder()
            .setDriver(BundledSQLiteDriver())
            .setQueryCoroutineContext(Dispatchers.IO)
            .build()
    }
```